### PR TITLE
Improve input field claim

### PR DESCRIPTION
**Pull request by biocom**
# Summary

**Account used:** `0x4bD39704a83dCB41B74b550988A832b83d5b9Cb0`

Tries the achieve:
- For the Airdrop row, hide the `Price` element.
- Making the investment amount input field editable, binding the value. Will probably need more work.
- The default value for the `value` attribute is always `0` (String) based on `investedAmount`. Prefer to return an empty string so it falls back to the `placeholder` attribute behavior.

## Note
- Crashes right now when entering a very big number
- the `max` attribute is currently not respected. Even when adding the `type="number"` attribute/value to the `Input` element. Probably something to do for the `onChange` handler.

https://user-images.githubusercontent.com/31534717/149932117-abb83096-a1fa-438f-9596-2641ff27f18d.mov

### CODE REVIEW

**Comment by W3stside** [comment]
actually why is max written in abbreviated form? can and should just be `Invest max possible` or `Invest maximum possible`

**Comment by biocom** [comment]
@W3stside The shorter the better (because of a variable `balance` width/size). But agreed, we could just rename `max.` to `max`

Will commit.

**Comment by elena-zh** [comment]
@biocom , I was also able to catch the app crash when I entered a period/comma into the input amount (decimal that starts with 0). Btw, impossible to enter '0' into the field
image
image

the same is when I enter letters. special symbols, etc.
image
image

Also, the app is crashed when enter an amount, then clear out the field.
image

**Comment by elena-zh** [comment]
Also, it is possible to input a negative amount into the field (I copied and pasted it)
image

**Comment by biocom** [comment]
@elena-zh 
> I'm not sure if it is related to this PR or not, but it would be great to disable the field and set its value to 100% when claim on behalf, as a user for this case will have to claim 100% amount only for an investment option.

Yes, this is a scenario where we need to enable the `disable` attribute on the `input` element.

**Review by anxolin — APPROVED** [review]
Approving!

I see there's some issues, however the two PRs michel is working on are mostly to tackle design. 

The error for input empty string for example, could be dealt in a future PR. Actually @nenadV91 is working in sth similar, so i think we can ignore for now. 

Same as investing on behalf of someone else, Elena you are right! although we will deal in another one.

File: src/custom/pages/Claim/ClaimsTable.tsx [hunk 1]
```diff
@@ -85,10 +85,11 @@ const ClaimsTableRow = ({
...
       <td>
-        <span>
-          Price:{' '}
-          <b>{isFree || !price ? '-' : `${formatSmart(price) || 0} vCoW per ${currencyAmount?.currency?.symbol}`}</b>
-        </span>
+        {!isFree || price ? (
+          <span>
+            Price: <b>{`${formatSmart(price) || 0} vCoW per ${currencyAmount?.currency?.symbol}`}</b>
+          </span>
+        ) : null}
```
Comment (alfetopito) [inline src/custom/pages/Claim/ClaimsTable.tsx:92]: Tiny refactoring suggestion:

```suggestion
        {!isFree || price && (
          <span>
            Price: <b>{`${formatSmart(price) || 0} vCoW per ${currencyAmount?.currency?.symbol}`}</b>
          </span>
        )}
```

**Review by alfetopito — APPROVED** [review]
Looks good

One small suggestion and this issue which will likely be fixed by Nenad, but just point out because it's kinda funny
Screen Shot 2022-01-18 at 10 21 31
The USDC amounts are probably not using the right amount of decimals :)
Or it's that issue with Rinkeby USDC used that has 18 rather than 6 decimals 🤷

**Pull request merged**